sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -514,6 +521,8 @@ public ConnectionConfiguration withApiKey(String apiKey) {
 
     /**
      * If Elasticsearch authentication is enabled, provide a bearer token.
+     * Be aware that you can only use one of {@Code withApiToken()}, {@code withBearerToken()} and
+     * {@code withDefaultHeaders} at the same time, as they will override eachother.
```

Comment:
Is there a reasonable way to add Precondition checks to try to alert the user when these mutually exclusive fields have been used in conjunction?

Reply:
There is but I didn't add it because the same issue was already lurking with just `withApiKey` and `withBearerToken`. I added checks to all three methods now to check that only one is used at a time - see my last update.